### PR TITLE
Add a few convproc functions and tests.

### PR DESCRIPTION
**Pull request by overfelt**
@jeff-cohere ,
I would like to get this branch merged into main so I could continue to add functions without bloating an already large merge request.  I would have already merged but for a problem with a couple of Skywalker validation tests which are commented out in the CMakeLists.txt file. I'm working with Balwinder to figure out why my results differ from the e3sm Skywalker python files and my best guess at this point is that the e3sm Skywalker files need to be regenerated.  

There are quite a few functions added, some of which could be parallel over layers within a column and some not.  I've added comments at the top of the functions for how parallelizable the function is:
* Some functions apply a non-trivial algorithm (non-associative, non-binary operator) to the previous layer to determine the values for the current layer so can not be parallel over the layers.
* Some functions use data from adjacent layers but in a read-only way and only set values on the current layer. These take a layer number, kk, as input but can be called in parallel.
* Some functions only access data in the current layer even though there is an input layer number, kk, as part of the function signature. The function signature could be changed to subscript the data outside of the function.

My plan is to determine how to call these over a Kokkos team object once the higher level functions that call these lower level functions are ported.

One thing to think about is how to call functions in other processes.  There are a few functions in wetdep that are used by convproc.   These should be stand-alone functions with simple signatures that are independent of any Atmosphere, Diagnostic or Prognostic classes. Once wetdep exists should convproc call them directly or should they be factored out into a separate translation unit?

### CODE REVIEW

**Comment by jeff-cohere** [comment]
Thanks, @overfelt . Reviewing now. Regarding e.g. wetdep--I would prefer that we keep wetdep functions in their own namespace for now (`mam4::wetdep`?) and let `convproc` call them directly. At some point, we should try to untangle these things, but I think we should do that deliberately when it's appropriate, and leave them tangled in a clear way for the moment. :-)

File: src/mam4xx/convproc.hpp [hunk 1]
```diff
@@ -79,16 +79,21 @@ class ConvProc {
...
     Config &operator=(const Config &) = default;
+
+    bool convproc_do_aer = true;
+    bool convproc_do_gas = false;
   };
 
   static constexpr int num_modes = AeroConfig::num_modes();
   static constexpr int num_aerosol_ids = AeroConfig::num_aerosol_ids();
+
+  // Constantants for MAM spciesi classes
```
Comment (jeff-cohere) [inline src/mam4xx/convproc.hpp:90]: some typos here

File: src/mam4xx/convproc.hpp [hunk 2]
```diff
@@ -1127,6 +1132,480 @@ void compute_wup(const int iconvtype, const int kk,
...
+   in :: mu_p_eudp[pver]      ! = mu_i[kp1] + eudp[k] [mb/s]
+   in :: aqfrac[pcnst_extd]   ! aqueous fraction of constituent in updraft [fraction]
+   in :: icwmr[pver]    ! Convective cloud water from zm scheme [kg/kg]
+   in :: rprd[pver]     ! Convective precipitation formation rate [kg/kg/s]
+   inout :: conu[pverp][pcnst_extd]   ! mix ratio in updraft at interfaces [kg/kg]
+   inout :: dconudt_wetdep[pverp][pcnst_extd] ! d(conu)/dt by wet removal[kg/kg/s]
+  */
+  // clang-format on
+  // cutoff value of cloud water for doing updraft [kg/kg]
+  const Real clw_cut = 1.0e-6;
```
Comment (jeff-cohere) [inline src/mam4xx/convproc.hpp:1560]: `BAD_CONSTANT`? Or at least something we should tag to revisit?

**Review by jeff-cohere — APPROVED** [review]
Merge away. Left a couple of very minor comments.

File: src/mam4xx/convproc.hpp (hunk 2 above)
Comment (overfelt) [reply src/mam4xx/convproc.hpp:1560]: Are we marking all non-attributed (no reference to a paper or anything else) constants as BAD_CONSTANT so we can revisit at a future date?

Comment (jeff-cohere) [reply src/mam4xx/convproc.hpp:1560]: I don't think we've been ultra-consistent about this. Certainly we mark physical constants this way. Error and "is there anything at all" thresholds less so (though given the lack of institutional knowledge, maybe we be). I'll leave this one to your judgement. The other threshold I saw in this PR is 1e-12, which to me is "really small in a numerical sense", so I'm fine not marking those.

**Pull request merged**